addons/point_of_sale/models/pos_order.py

```diff
@@ -189,13 +189,12 @@ def _process_payment_lines(self, pos_order, order, pos_session, draft):
             order.add_payment(return_payment_vals)
 
     def _prepare_invoice_line(self, order_line):
-        name = order_line.product_id.default_code + " " + order_line.product_id.display_name if order_line.product_id.default_code else order_line.product_id.display_name
```

Comment:
Can you confirm that even if we remove this logic, the display_name will always have the default_code?

Reply:
It was not showing the default code, I modified the code to keep it